src/bin/electrs.rs

```diff
@@ -122,8 +125,29 @@ fn run_server(config: Arc<Config>) -> Result<()> {
             tip = current_tip;
         };
 
+        // FIXME(jamesdorfman): couldn't figure out how to import it from util
+        let log_fn_duration = |fn_name: &str, duration: u128| {
+            let log = json!({
+                "fn_name": fn_name,
+                "duration_micros": duration,
+            });
+            println!("{}", log);
+        };
+
         // Update mempool
-        mempool.write().unwrap().update(&daemon)?;
+
+        let t = Instant::now();
+
+        let old_txids = mempool.read().unwrap().old_txids();
```

Comment:
This mempool syncing logic should be in `mempool.rs` and not as part of the `electrs` bin.

I would suggest adding a static method to `Mempool` that takes an `RwLock<Mempool>` parameter and locks it internally.